modules/init-snippet-attach-ebs-volume/instance_id.tpl

```diff
@@ -0,0 +1,7 @@
+if which wget; then
+  INSTANCE_ID="$(wget -O- http://169.254.169.254/latest/meta-data/instance-id)"
+elif which curl; then
+  INSTANCE_ID="$(curl http://169.254.169.254/latest/meta-data/instance-id)"
+fi
+
```

Comment:
If no curl and no wget, is this a silent failure with redherring elsewhere, or do we fail more loudly?

Reply:
This is handled in snippet.tpl. But I will move it here.